helix-core/src/main/java/org/apache/helix/controller/stages/PerReplicaThrottleStage.java

```diff
@@ -0,0 +1,951 @@
+package org.apache.helix.controller.stages;
```

Comment:
This new file is hard to review... Can we do this? Since most of the code is copied from IntermediateStage. Let's use IntermediateStage file and make the change for the review first as a PR? Then after reviewed, we can rename the file to PerReplicaThrottleStage.

I know split it into two PRs could break tests for first one. But I think that could be more efficient for reviewing.

Reply:
Let me copied the code to Intermedidate stage for you to have a look at the difference. This won't compile though. 

That said, the code is not really copied from IntermediateStage. There is not one to one method. Be warned, there will be probably way too many differences.